src/entry.rs

```diff
@@ -689,22 +690,29 @@ impl<R: Read + Unpin> EntryFields<R> {
                     }
                 }
             }?;
+
+            let size = usize::try_from(self.size).unwrap_or(usize::MAX);
+            let capacity = cmp::min(size, 128 * 1024);
+            let mut writer = io::BufWriter::with_capacity(capacity, &mut f);
             for io in self.data.drain(..) {
                 match io {
                     EntryIo::Data(mut d) => {
                         let expected = d.limit();
-                        if io::copy(&mut d, &mut f).await? != expected {
+                        if io::copy(&mut d, &mut writer).await? != expected {
                             return Err(other("failed to write entire file"));
                         }
                     }
                     EntryIo::Pad(d) => {
                         // TODO: checked cast to i64
-                        let to = SeekFrom::Current(d.limit() as i64);
-                        let size = f.seek(to).await?;
-                        f.set_len(size).await?;
+                        let pad_len = d.limit() as i64;
+                        writer.flush().await?;
+                        let f = writer.get_mut();
+                        let new_size = f.seek(SeekFrom::Current(pad_len)).await?;
+                        f.set_len(new_size).await?;
```

Comment:
Is there documentation on when we need this padding and how the seek/set_len trick works?